### ISSUE TITLE
App launcher?

### ISSUE THREAD

**Issue by MariaSolOs** [pagankeymaster/vile#31]
Amazing repo! It's been a great reference as I build my own `eww` widgets :)

I do want to ask: Is the app launcher featured in the screenshots part of your `eww` setup? Or are you using something else like `rofi`?

**Comment by dharmx** [comment]
Hello.

Yes, the launcher is ROFI. It can be found in my dotfiles.
Although, I don't quite remember if I have removed it or, not
(migrated to AwesomeWM).

EDIT: Nevermind. It is still [there](https://github.com/dharmx/dharmx/tree/main/config/_ui/rofi). Credit to [joni](https://www.reddit.com/r/unixporn/comments/ucjfha/rofi_side/).

Thank you!

**Comment by MariaSolOs** [comment]
Thanks @dharmx!

**Comment by MariaSolOs** [comment]
Out of curiosity (and if you don't mind me asking): Why did you switch to AwesomeWM?

**Comment by dharmx** [comment]
Mainly because AwesomeWM is a framework window manager. That roughly means that it has the tools for building your own Desktop Environment. For example you have:

- A builtin keyboard daemon (like sxhkd).
- A builtin notification daemon (like mako/dunst).
- A builtin bar.
- Titlebars, etc.

And, all of this can be configured using Lua. And, they are not segmented i.e. you can call in titlebar libs within your keyboard daemon. That is for instance you can remove/add titlebars on `<C-T>` or, automatically disable titlbars when a window is maximized.

Secondly, since this is using Lua/LuaJIT we can use Lua-transpiled languages as well like Yuescript, Teal, PurescriptToLua, typescriptToLua, etc.

Thirdly, you can call in Lua libraries using luarocks. And, you can use some AwesomeWM libraries as well like the animation library [Rubato](https://github.com/andOrlando/rubato/), a custom layout making lib [layout-machi](https://github.com/xinhaoyuan/layout-machi), a pre-made widgets library (like vile) [bling](https://blingcorp.github.io/bling/#/). And, much more.

TL;DR: A LOOOOOT of things to play with. AwesomeWM sub-functions integrates really well with each other. It uses a well-known and very fast scripting language and your world won't be revolving around parsing STDOUT.

**Comment by dharmx** [comment]
Also, you might like [ags](https://github.com/Aylur/ags). It claims to have improved on eww's shortcomings and uses typescript as well.

**Comment by MariaSolOs** [comment]
Oh that's very insightful! I'm currently using Hyprland but I'm not sure I actually need Wayland rn and considering other alternatives. I'll check out AwesomeWM!
